LibSWBF2/Wrappers/Terrain.cpp

```diff
@@ -118,6 +118,70 @@ namespace LibSWBF2::Wrappers
 		return true;
 	}
 
+
+	void Terrain::GetBlendMap(uint32_t& dim, uint32_t& elementSize, uint8_t*& finalBlendMapData) const
+	{
+		auto *info = p_Terrain -> p_Info;
+		
+		uint16_t& gridSize = info->m_GridSize;
+		float_t& gridUnitSize = info->m_GridUnitSize;
+		uint16_t numVertsPerPatchEdge = info->m_PatchEdgeSize;
+		uint16_t dataEdgeSize = numVertsPerPatchEdge + 1;
+
+		uint16_t numPatchesPerRow = gridSize / numVertsPerPatchEdge;
+		uint32_t numTexLayers = (uint32_t) info -> m_TextureCount;
+
+		int dataLength = (int) (gridSize * gridSize * numTexLayers);
+        
+        finalBlendMapData = new uint8_t[dataLength]();
+        dim = gridSize;
+		elementSize = numTexLayers; 
+
+
+		List<PTCH*>& patches = p_Terrain->p_Patches->m_Patches;
+
+		for (size_t i = 0; i < patches.Size(); i++)
+		{	
+			auto *curPatch = patches[i];
+			auto *patchInfo = curPatch -> p_PatchInfo;
+			auto *patchSplatChunk = curPatch -> m_TextureBuffer;
+			uint8_t *curPatchBlendMap = patchSplatChunk -> p_SplatMapData;
+
+            List<uint32_t>& slotsList = patchInfo -> m_TextureSlotsUsed;
+            int numSlotsInPatch = slotsList.Size();
+
+			int globalPatchY = i / (int) numPatchesPerRow;
+			int globalPatchX = i % (int) numPatchesPerRow;
+
+			int patchStartIndex = globalPatchY * numVertsPerPatchEdge * numPatchesPerRow * numVertsPerPatchEdge + globalPatchX * numVertsPerPatchEdge;
+
+			for (int j = 0; j < patchSplatChunk -> m_ElementCount; j++)
+			{
+				int localPatchY = j / dataEdgeSize; 
+				int localPatchX = j % dataEdgeSize;
+
+				//Starting index into final array (finalBlendMapData)
+				int globalDataIndex = numTexLayers * (patchStartIndex + localPatchY * numVertsPerPatchEdge * numPatchesPerRow + localPatchX);
+				
+				//Index into current patch's VBUF's blend data (curPatchBlendMap)
+				int localPatchIndex = numSlotsInPatch * (localPatchY * dataEdgeSize + localPatchX);
+
+				for (int k = 0; k < numSlotsInPatch; k++)
+				{
+					int slot = (int) slotsList[k];
+					int finalDataIndex = globalDataIndex + slot;
+
+					if (finalDataIndex < dataLength)
+					{	
+						finalBlendMapData[finalDataIndex] = (uint8_t) curPatchBlendMap[localPatchIndex + k];
+					}
+				}
+			}
+		} 
+		    
+	}
+
```

Comment:
I suppose this could've been part of FromChunks to cut some duplication, but I wanted to do it from scratch to understand the actual structure

Reply:
Sure thing. I mean we can put it into FromChunk afterwards if the calc penalty turns out pretty high.